### PR TITLE
Add build script and method to save data in CSV format. 

### PR DESCRIPTION
**Pull request by dfquaresma**
This PR implements a simple script to execute "mvn clean" and "mvn install" in all projects, working as a build project. It also adds a method used to save the log of shed data in CSV format. This last addition requires a enviroment variable to set the name of the CSV shed file. This file is beeing saved at the same directory as the application is called.

### CODE REVIEW

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java [hunk 1]
```diff
@@ -11,6 +13,7 @@
 	private static final Duration WAIT_FOR_TRAILERS_SLEEP_MILLIS = Duration.ofMillis(10);
 	private static final int SAMPLE_HISTORY_SIZE = 5;
 	private static final boolean DEBUG_GCI = Boolean.parseBoolean(System.getenv("debug_gci"));
+	private String csv_file_name;
```
Comment (danielfireman) [inline core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: csvFileName

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java [hunk 2]
```diff
@@ -37,6 +40,12 @@ public GarbageCollectorControlInterceptor(Heap monitor, Executor executor) {
 		this.finished = new AtomicLong();
 		this.shedRequests = new AtomicLong();
 		this.sheddingThreshold = new SheddingThreshold();
+		
+		if (DEBUG_GCI) {
+			csv_file_name = System.getenv("csv_file_name");
```
Comment (danielfireman) [inline core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Environment variables capitalized, please.  Also, CSV is too generic. Lets make it more strict.

SHED_RATIO_CSV_FILE

That said, you don´t need the DEBUG_GCI flag anymore. If the SHED_RATIO_CSV_FILE is set, you write this out.

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java [hunk 3]
```diff
@@ -58,6 +67,30 @@ private ShedResponse shed() {
...
+	public void saveInCSV(String line) {
+		FileWriter fw = null;
+		BufferedWriter bw = null;
+
+		try {
+			fw = new FileWriter(csv_file_name, true);
+			bw = new BufferedWriter(fw);
+			bw.write(line + System.lineSeparator());
+			bw.flush();
+
```
Comment (danielfireman) [inline core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Remove blank line.

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java [hunk 4]
```diff
@@ -58,6 +67,30 @@ private ShedResponse shed() {
 		return new ShedResponse(true);
 	}
 
+	public void saveInCSV(String line) {
+		FileWriter fw = null;
+		BufferedWriter bw = null;
+
+		try {
+			fw = new FileWriter(csv_file_name, true);
```
Comment (danielfireman) [inline core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Did you check what happen when you open and close the filewriter and bufferedWriter at every write. This sounds super slow to me, thus I believe that we are going to have an performance impact on our experiments.

Could you please check if it is worth to optmitze?

Comment (danielfireman) [reply core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Example of library and some other compared libraries.

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java [hunk 5]
```diff
@@ -58,6 +67,30 @@ private ShedResponse shed() {
 		return new ShedResponse(true);
 	}
 
+	public void saveInCSV(String line) {
+		FileWriter fw = null;
+		BufferedWriter bw = null;
+
```
Comment (danielfireman) [inline core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Remove blank line.

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java [hunk 6]
```diff
@@ -58,6 +67,30 @@ private ShedResponse shed() {
...
+		try {
+			fw = new FileWriter(csv_file_name, true);
+			bw = new BufferedWriter(fw);
+			bw.write(line + System.lineSeparator());
+			bw.flush();
+
+		} catch (Exception e) {
+			e.printStackTrace();
```
Comment (danielfireman) [inline core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: To think: Is this what we want?

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java (hunk 3 above)
Comment (dfquaresma) [reply core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Done

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java (hunk 5 above)
Comment (dfquaresma) [reply core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Done

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java (hunk 4 above)
Comment (dfquaresma) [reply core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Gonna check it out.

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java (hunk 2 above)
Comment (dfquaresma) [reply core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Done

File: core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java [hunk 7]
```diff
@@ -58,12 +88,20 @@ private ShedResponse shed() {
 		return new ShedResponse(true);
 	}
 
+	private void saveInCSV(String line) {
```
[the remark this answers is not in the document]
Comment (danielfireman) [reply core/src/main/java/com/gcinterceptor/core/GarbageCollectorControlInterceptor.java]: Also pass the bufferedWriter, this couls be a static methos.

**Pull request merged**